File: 206-ReverseLinkedList/206-ReverseLinkedList.c
// Last Updated: 2/5/2026, 9:24:23 AM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
 typedef struct ListNode node;
struct ListNode* reverseList(struct ListNode* head) {
    node *pn=NULL;
    node *cn=head;
    node *nn=head;
    while(cn!=NULL){
        nn=nn->next;
        cn->next=pn;
        pn=cn;
        cn=nn;  
    }
    return pn;
    
}

File: 206-ReverseLinkedList/206-ReverseLinkedList.py
# Last Updated: 2/5/2026, 9:24:17 AM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None:
            return head
        pn=None
        cn=head
        nn=head.next
        while cn:
            cn.next=pn
            pn=cn
            cn=nn
            if nn:
                nn=nn.next
        return pn